src/openMVG/multiview/solver_resection_p4pf.cpp

```diff
@@ -0,0 +1,1338 @@
+  M[1397] = coefs[39];
+  M[1484] = coefs[39];
+  M[3241] = coefs[39];
+  M[3326] = coefs[39];
+  M[3413] = coefs[39];
+  M[3500] = coefs[39];
+  M[3674] = coefs[39];
+  M[3761] = coefs[39];
+  M[3848] = coefs[39];
+  M[3935] = coefs[39];
+  M[5602] = coefs[39];
+  M[5689] = coefs[39];
+  M[5776] = coefs[39];
+  M[5860] = coefs[39];
+  M[5946] = coefs[39];
+  M[6032] = coefs[39];
+  M[6117] = coefs[39];
+  M[6724] = coefs[39];
+  M[6807] = coefs[39];
+  M[350] = coefs[40];
+  M[1226] = coefs[40];
+  M[1313] = coefs[40];
+  M[1400] = coefs[40];
+  M[1487] = coefs[40];
+  M[3242] = coefs[40];
+  M[3329] = coefs[40];
+  M[3416] = coefs[40];
+  M[3503] = coefs[40];
+  M[3590] = coefs[40];
+  M[3677] = coefs[40];
+  M[3764] = coefs[40];
+  M[3851] = coefs[40];
+  M[5605] = coefs[40];
+  M[5692] = coefs[40];
+  M[5779] = coefs[40];
+  M[5864] = coefs[40];
+  M[5950] = coefs[40];
+  M[6037] = coefs[40];
+  M[6122] = coefs[40];
+  M[6728] = coefs[40];
+  M[6812] = coefs[40];
+  M[351] = coefs[41];
+  M[1230] = coefs[41];
+  M[1317] = coefs[41];
+  M[1404] = coefs[41];
+  M[1491] = coefs[41];
+  M[3250] = coefs[41];
+  M[3337] = coefs[41];
+  M[3424] = coefs[41];
+  M[3511] = coefs[41];
+  M[3598] = coefs[41];
+  M[3685] = coefs[41];
+  M[3772] = coefs[41];
+  M[3859] = coefs[41];
+  M[5617] = coefs[41];
+  M[5704] = coefs[41];
+  M[5791] = coefs[41];
+  M[5876] = coefs[41];
+  M[6050] = coefs[41];
+  M[6135] = coefs[41];
+  M[6742] = coefs[41];
+  M[6826] = coefs[41];
+
+  // GJ elimination
+  GJ(M, 78, 88, 2.2204e-11);
+
+  // action matrix
+  memset(A, 0, sizeof (double) * 100);
+
+  A[1] = 1;
+  A[15] = 1;
+  A[26] = 1;
+  A[37] = 1;
+  A[48] = 1;
+  A[50] = -M[6599];
+  A[51] = -M[6598];
+  A[52] = -M[6597];
+  A[53] = -M[6596];
+  A[54] = -M[6595];
+  A[55] = -M[6594];
+  A[56] = -M[6593];
+  A[57] = -M[6592];
+  A[58] = -M[6591];
+  A[59] = -M[6590];
+  A[60] = -M[6511];
+  A[61] = -M[6510];
+  A[62] = -M[6509];
+  A[63] = -M[6508];
+  A[64] = -M[6507];
+  A[65] = -M[6506];
+  A[66] = -M[6505];
+  A[67] = -M[6504];
+  A[68] = -M[6503];
+  A[69] = -M[6502];
+  A[70] = -M[6423];
+  A[71] = -M[6422];
+  A[72] = -M[6421];
+  A[73] = -M[6420];
+  A[74] = -M[6419];
+  A[75] = -M[6418];
+  A[76] = -M[6417];
+  A[77] = -M[6416];
+  A[78] = -M[6415];
+  A[79] = -M[6414];
+  A[80] = -M[6335];
+  A[81] = -M[6334];
+  A[82] = -M[6333];
+  A[83] = -M[6332];
+  A[84] = -M[6331];
+  A[85] = -M[6330];
+  A[86] = -M[6329];
+  A[87] = -M[6328];
+  A[88] = -M[6327];
+  A[89] = -M[6326];
+  A[90] = -M[6247];
+  A[91] = -M[6246];
+  A[92] = -M[6245];
+  A[93] = -M[6244];
+  A[94] = -M[6243];
+  A[95] = -M[6242];
+  A[96] = -M[6241];
+  A[97] = -M[6240];
+  A[98] = -M[6239];
+  A[99] = -M[6238];
+}
+
+bool isNan(Eigen::MatrixXcd *A)
+{
+  Mat B = A->real();
+  for(int i = 0; i < B.cols() * B.rows(); i++)
+  {
+    if(isnan(B.data()[i])) return true;
+  }
+  return false;
+}
+
+bool validSol(Eigen::MatrixXcd *sol, Mat *vSol)
+{
+  Mat imSol = sol->imag();
```

Comment:
add an assert to check the dimensions of sol, like
assert(sol.cols == 10 && sol.rows == 4)